### PR TITLE
DB Inventory & Skill Tree

### PR DESCRIPTION
**Pull request by bwu113**
Initial PR Commit:

Added Inventory Table
Added Item Table
Added Skill Tree Table

Update 2:

Renamed InventoryTab -> Inventory,
Added columns to Inventory Table,
Added [Tab] Column to Item Table,
Added the ability to remove accounts,
Added the ability to retrieve account by username,
Added the ability to remove characters,
Modified AddCharacter in CharacterManager to create Inventory along with character,
Added InventoryManager (With incomplete functionality)

Update 3:

Changed SetAccInfo -> EditAccInfo,
Changed SetCharInfo -> EditCharInfo,
InventoryManager functionality is completed,
SkillTreeManager functionality is completed,
Created Uid column in SkillTree Table for easier querying.

Update 4:

Changed variable names and parameter names for Controller Classes.

### CODE REVIEW

File: Ms2Database/Controllers/AccountManager.cs [hunk 1]
```diff
@@ -33,7 +43,16 @@ public void CreateAccount(string username, string password)
             }
         }
 
-        public void SetAccInfo(long id, string username = "", string password = "") // Allows account entry changes
+        public Account GetAccInfoByUser(string username)
+        {
+            using (Ms2DbContext context = new Ms2DbContext())
+            {
+                Account account = context.Accounts.FirstOrDefault(a => a.Username.ToLower() == username.ToLower()); // Retrieve entry by Id
```
Comment (AlanMorel) [inline Ms2Database/Controllers/AccountManager.cs:50]: By Id?

File: Ms2Database/Controllers/CharacterManager.cs [hunk 2]
```diff
@@ -42,11 +57,11 @@ public Character GetCharInfo(long charId)
             }
         }
 
-        public void SetCharInfo(Character CharIn)
+        public void EditCharInfo(Character character)
         {
             using (Ms2DbContext context = new Ms2DbContext())
             {
-                Character character = CharIn;
+                Character OriginCharacter = character;
```
Comment (AlanMorel) [inline Ms2Database/Controllers/CharacterManager.cs:64]: why this naming convention?

**Review by AlanMorel — CHANGES_REQUESTED** [review]

File: Ms2Database/Controllers/AccountManager.cs (hunk 1 above)
Comment (bwu113) [reply Ms2Database/Controllers/AccountManager.cs:50]: Oh snap. Nice catch let me fix that

File: Ms2Database/Controllers/CharacterManager.cs (hunk 2 above)
Comment (bwu113) [reply Ms2Database/Controllers/CharacterManager.cs:64]: Figured I'd name the this.variable to something more fitting since param is the same name

Comment (bwu113) [reply Ms2Database/Controllers/CharacterManager.cs:64]: Prob change it to something different

**Review by AlanMorel — APPROVED** [review]

**Review by angelotadeucci — COMMENTED** [review]
Looks good to me.

**Pull request merged**